src/core/memblock.h

```diff
@@ -0,0 +1,17 @@
+#ifndef CORE_MEM_BLOCK_H
+#define CORE_MEM_BLOCK_H
+
+#include <stdint.h>
+
+typedef struct {
+	char *mem;
+	uint32_t size;
```

Comment:
`unsigned int`should work just as well without requiring an extra include. It's what is used in `core/array` as well I think.

Reply:
I’d prefer to keep uint32_s as it’s clearer and more concise. Unsigned int can be 2 or 4 bytes so it’s a bit of a gamble type.

Reply:
It can only be 2 bytes on 16-bit hardware I think or very old compilers, which has no business running Augustus.

It's a personal thing of mine. I like to avoid `stdint.h` when possible.

Reply:
Ok. Any particular reason why? Seems way better to me.

Reply:
Code size? I don't like to add custom code (even from includes) unless it's absolutely needed, like on file buffers. But that's me.

`unsigned long` works just fine too, and it'll be 4 bytes.

Reply:
Yeah…I think we’re going to have to agree to disagree on this one. C’s native integer types are too verbose and inconsistent for me. Is this a dealbreaker for you?

Reply:
What about `size_t`?

Reply:
Had to read about it a bit but it does seem somewhat appropriate given it was designed for array counting and such. I still don’t like that it’s variable sized and its minimum required size is smaller than what we allocate by default for the compression buffer, but at least it’s not verbose so I think I’m ok with it.